### PR TITLE
[ST-Engine][Starter] Add seatunnel own engine starter and e2e

### PR DESCRIPTION
**Pull request by Hisoka-X**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/utils/CommandLineUtils.java [hunk 1]
```diff
@@ -43,10 +42,10 @@ public static FlinkCommandArgs parseFlinkArgs(String[] args) {
         return flinkCommandArgs;
     }
 
-    public static FlinkCommandArgs parseCommandArgs(String[] args, FlinkJobType jobType) {
+    public static FlinkCommandArgs parseCommandArgs(String[] args) {
         FlinkCommandArgs flinkCommandArgs = new FlinkCommandArgs();
         JCommander jCommander = JCommander.newBuilder()
-            .programName(jobType.getType())
+            .programName("start-seatunnel-flink-connector-v2.sh")
```
Comment (hailin0) [inline seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/utils/CommandLineUtils.java:48]: Need to update [this](https://github.com/apache/incubator-seatunnel/pull/2675/files#diff-d4c3677ccf36bf18f8b9fde06fbbca8c3b686d43afb7c65d19105069f142843dR34) when merging into dev later

File: seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/SparkStarter.java [hunk 2]
```diff
@@ -134,7 +134,7 @@ static SparkStarter getInstance(String[] args) {
     private static SparkCommandArgs parseCommandArgs(String[] args) {
         SparkCommandArgs commandArgs = new SparkCommandArgs();
         JCommander commander = JCommander.newBuilder()
-                .programName("start-seatunnel-spark.sh")
+                .programName("start-seatunnel-spark-connector-v2.sh")
```
Comment (hailin0) [inline seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/SparkStarter.java:137]: as above

File: seatunnel-e2e/seatunnel-engine-e2e/pom.xml [hunk 3]
```diff
@@ -27,6 +27,12 @@
     <modelVersion>4.0.0</modelVersion>
 
     <artifactId>seatunnel-engine-e2e</artifactId>
+    <packaging>pom</packaging>
+
+    <modules>
+        <module>seatunnel-engine-e2e-base</module>
+        <module>seatunnel-engine-connector-console-e2e</module>
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-engine-e2e/pom.xml]: module name style
 
reference
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-flink-connector-v2-e2e
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-spark-connector-v2-e2e

File: seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-connector-console-e2e/src/test/resources/fakesource_to_console.conf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+transform {
+}
+
+sink {
+  console {
+  }
+
+  # If you would like to get more information about how to configure seatunnel and see full list of sink plugins,
+  # please go to https://seatunnel.apache.org/docs/flink/configuration/sink-plugins/Console
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-connector-console-e2e/src/test/resources/fakesource_to_console.conf]: remove

File: seatunnel-engine/seatunnel-engine-client/pom.xml [hunk 5]
```diff
@@ -68,7 +68,7 @@
         </dependency>
         <dependency>
             <groupId>org.apache.seatunnel</groupId>
-            <artifactId>connector-file-local</artifactId>
+            <artifactId>connector-console</artifactId>
```
Comment (hailin0) [inline seatunnel-engine/seatunnel-engine-client/pom.xml]: St-client depends on connector feeling strange

Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-client/pom.xml]: just for test in client test module

File: seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-connector-console-e2e/src/test/resources/fakesource_to_console.conf (hunk 4 above)
Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-connector-console-e2e/src/test/resources/fakesource_to_console.conf]: Done

File: seatunnel-e2e/seatunnel-engine-e2e/pom.xml (hunk 3 above)
Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-engine-e2e/pom.xml]: Done

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/utils/CommandLineUtils.java (hunk 1 above)
Comment (Hisoka-X) [reply seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/utils/CommandLineUtils.java:48]: OK

File: seatunnel-core/README.md [hunk 6]
```diff
@@ -5,4 +5,7 @@ This module is the seatunnel job entrypoint. Seatunnel jobs are started by the b
 
 - seatunnel-core-flink: The flink job starter.
 - seatunnel-core-flink-sql: The flink sql job starter.
-- seatunnel-core-spark: The spark job starter.
\ No newline at end of file
+- seatunnel-core-spark: The spark job starter.
+- seatunnel-spark-starter: The spark job starter for connector-v2.
+- seatunnel-spark-starter: The flink job starter for connector-v2.
```
Comment (EricJoy2048) [inline seatunnel-core/README.md]: `seatunnel-spark-starter` ?

File: seatunnel-core/README.md [hunk 7]
```diff
@@ -5,4 +5,7 @@ This module is the seatunnel job entrypoint. Seatunnel jobs are started by the b
 
 - seatunnel-core-flink: The flink job starter.
 - seatunnel-core-flink-sql: The flink sql job starter.
-- seatunnel-core-spark: The spark job starter.
\ No newline at end of file
+- seatunnel-core-spark: The spark job starter.
```
Comment (EricJoy2048) [inline seatunnel-core/README.md:8]: `seatunnel-core-spark: The spark job starter` ?

**Review by ashulin — COMMENTED** [review]
Can you split this into two PR? Since this involves a lot of non-engine modules, I would expect the changes to the starter module to be committed to the `dev` branch first.

**Comment by Hisoka-X** [comment]
> Can you split this into two PR? Since this involves a lot of non-engine modules, I would expect the changes to the starter module to be committed to the `dev` branch first.

OK

**Comment by EricJoy2048** [comment]
> > Can you split this into two PR? Since this involves a lot of non-engine modules, I would expect the changes to the starter module to be committed to the `dev` branch first.
> 
> OK

https://github.com/apache/incubator-seatunnel/pull/2726 has been merged.

File: seatunnel-core/README.md (hunk 7 above)
Comment (Hisoka-X) [reply seatunnel-core/README.md:8]: Yes, any suggestion?

Comment (EricJoy2048) [reply seatunnel-core/README.md:8]: Got it. 
From the name `seatunnel-core-spark` I can not know it's a starter. But now we don't care about it.

Comment (EricJoy2048) [reply seatunnel-core/README.md:8]: > Yes, any suggestion?

Got it. 
From the name `seatunnel-core-spark` I can not know it's a starter. But now we don't care about it.

**Pull request merged**